### PR TITLE
Re-adds encoder list to text features documentation, various sequence and text feature updates

### PR DESCRIPTION
**Pull request by dantreiman**
TL;DR:  don't make users go to the sequence features page to understand text features.

Text features and sequence features share most of their basic encoders.  Previously, we left a 'refer to sequence features doc' link.  I've change my mind on this.  It would be better to describe encoders on both sequence and text pages: if a user is reading the text features doc, all information about encoders they need should be on that page, both for convenience and for search.

Also:
- Removed parameter section of each HF encoder.  Added short descriptions to most of them.
- Re-added decoder section for text features.
- Added `mt5` and `transformer_xl` 
- Minor wording changes and edits
- Re-structures metrics of sequence and text features to a list

### CODE REVIEW

File: docs/configuration/features/text_features.md [hunk 1]
```diff
@@ -47,10 +58,11 @@ preprocessing:
 Text input feature parameters are
 
 - `encoder` (default `parallel_cnn`): encoder to use for the input text feature. The available encoders include encoders
-used for [Sequence Features](../sequence_features#sequence-input-features-and-encoders) as well as encoders from the huggingface
-transformers library: `bert`, `gpt`, `gpt2`, `xlnet`, `xlm`, `roberta`, `distilbert`, `ctrl`, `camembert`, `albert`,
-`t5`, `xlmroberta`, `flaubert`, `electra`, `longformer` and `auto-transformer`.
-- `tied` (default `null`): name of the input feature to tie the weights of the encoder with. It needs to be the name of a feature of the same type and with the same encoder parameters.
+used for [Sequence Features](../sequence_features#sequence-input-features-and-encoders) as well as encoders from the
+huggingface transformers library: `bert`, `gpt`, `gpt2`, `xlnet`, `xlm`, `roberta`, `transformer_xl`, `distilbert`,
+`ctrl`, `camembert`, `albert`, `t5`, `mt5`, `xlmroberta`, `flaubert`, `electra`, and `longformer`.
```
Comment (justinxzhao) [inline docs/configuration/features/text_features.md:63]: Could we also add `auto_transformer`?

**Review by justinxzhao — APPROVED** [review]
This looks great! I agree with the duplication so that each page can stand on its own.

Hopefully with schemas and automatic docs generation we'll have a scalable way to do this.

Noting that we should be mindful of these updated wordings when we back-port the parameter documentation spreadsheet + these docs to registered schemas and docstrings.

File: docs/configuration/features/text_features.md (hunk 1 above)
Comment (dantreiman) [reply docs/configuration/features/text_features.md]: Good call.  Done!
